lib/CssColors/Ecto/color.ex

```diff
@@ -47,14 +52,36 @@ defmodule CssColors.Ecto.Color do
     end
   end
 
-  @spec dump(CssColors.color) :: {:ok, String.t}
+  @spec dump(CssColors.color()) :: {:ok, String.t()}
   def dump(color = %CssColors.RGB{}), do: {:ok, do_dump(color)}
   def dump(color = %CssColors.HSL{}), do: {:ok, do_dump(color)}
   def dump(_), do: :error
 
   defp do_dump(color) do
-    round_to = 4
-    to_string CssColors.adjust(color, round_to, :alpha, fn(value, round_to) -> Float.round(value, round_to) end)
-
+    to_string(
+      CssColors.adjust(color, @round_to, :alpha, fn value, round_to ->
+        Float.round(value, @round_to)
+      end)
+    )
   end
+
+  @spec equal?(CssColors.color(), CssColors.color()) :: boolean()
+  def equal?(a, b), do: check_equal?(normalize(a), normalize(b))
+
+  defp normalize(color = %CssColors.RGB{}),
+    do: CssColors.adjust(color, @round_to, :alpha, &Float.round(&1, &2))
+
+  defp normalize(color = %CssColors.HSL{}), do: normalize(CssColors.rgb(color))
+  defp normalize(_), do: nil
```

Comment:
I don't think we want this catch-all here. If we do send a values that is not a color struct, we should fail.

Reply:
Oh, what about:


```
  defp do_dump(color),
    do: color |> round |> to_string

  defp normalize(color) do
    color
    |> CssColors.rgb
    |> CssColors.adjust(@round_to, :alpha, &Float.round(&1, &2))

  defp round(color),
    do: CssColors.adjust(color, @round_to, :alpha, &Float.round(&1, &2))
```

to make it clear that the rounding is shared between the serialization and the comparison?

Reply:
I've updated with the rounding-change, great suggestion!

Regarding the catch-all, Ecto [calls](https://github.com/elixir-ecto/ecto/blob/master/lib/ecto/changeset.ex#L567) `equal?` with the new and the old value where either can be `nil` so I think it needs to stay.